improve docs user story for members to join orgs

s-santillan opened the pull request:
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] **no todos or tks remain for this PR**
- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link


* Initially, Raj brought up the need to have docs specifically for members to be "invited" (or as close to "invited" as we can make them feel).
* While we don't have an explicit invite feature yet, we do have live improvements to a member's first sign in that haven't been documented.
* When I reviewed the docs, they are out of sync with the product in the sense that the current docs focus solely on the first-person-in-an-org to use Semgrep and didn't accommodate "team members". **With this update, orgs and team members are better accommodated**.
* I opted to rewrite the doc focusing on **guiding** both user paths, with **steps** rather than conceptual info that can be read elsewhere.

## Screenshots

<img width="1080" alt="2023-08-22_02-24-29" src="https://github.com/returntocorp/semgrep-docs/assets/20766840/8c681594-0672-4303-9248-7aff730d6d59">

<img width="1080" alt="2023-08-22_02-24-49" src="https://github.com/returntocorp/semgrep-docs/assets/20766840/2eb72c39-76b9-409d-8c9b-d7a1f70eb298">

sachishah commented:
@s-santillan , for Joining an Org, should we link out to the SSO instructions?

For scanning, we're making several improvements to zero-config scanning (goal is to get it to a Public Beta state by the end of the week) as well as the recommended path for new users — should we hold off on the changes here until the product updates are live over the next two weeks?

armchairlinguist commented on docs/pricing-and-billing.mdx:
```diff
@@ -28,7 +28,7 @@ Semgrep's various functionalities are available through several offerings:
 <dd>The OSS (open source software) offering includes Semgrep OSS Engine, a fast static analysis command line tool for finding bugs and and enforcing code standards. It is licensed under LGPL 2.1. </dd>
 <dt>Semgrep Team tier</dt>
-<dd>The Team tier includes Semgrep Code and Semgrep Supply Chain (SSC), which enables users to scan both first-party and third-party open source dependencies. It also includes Semgrep Cloud Platform, a web app enabling users to manage users, organizations, repositories, and scans.</dd>
+<dd>The Team tier includes Semgrep Code and Semgrep Supply Chain (SSC), which enables users to scan both first-party and third-party open source dependencies. It also includes Semgrep Cloud Platform, a web app enabling users to manage users, organizations, repositories, and scans. Team tier also includes <strong>cross-function (intrafile)</strong> and <strong>cross-file (interfile) analysis</stro ...
```
Can we turn the inclusions here into a bullet list? I _think_ that's allowed within a definition list, and it would be better now that there are three:

The Team tier includes:
* Semgrep Code and Semgrep Supply Chain (SSC), which enable users to scan both first-party and third-party open source dependencies. 
* <strong>Cross-function (intrafile)</strong> and <strong>cross-file (interfile) analysis</strong> through <a href="/semgrep-code/semgrep-pro-engine-intro/">Semgrep Pro Engine</a>.
* Semgrep Cloud Platform, a web app enabling users to manage users, organizations, repositories, and scans.

s-santillan replied:
Yep, resolved in latest :)

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,82 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+## Joining an organization
+
+:::info Prerequisites
+- Your admin or inviter should have an existing organization (org) in Semgrep that is connected to the org in either GitHub or GitLab.
```
```suggestion
- Your admin or inviter should have an existing organization (org) in Semgrep that is connected to an org in either GitHub or GitLab.

```

armchairlinguist replied:
"Inviter" is a bit weird here (funny since we use "invitee" a lot, but I don't hear "inviter" much). I might go with "Your admin or the person inviting you to SCP".

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,82 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+- You must be a member of the GitHub or GitLab org that is connected to Semgrep.
+:::
+
+For **members to join an organization**, perform the following steps to be added to the org:
```
```suggestion
For **members**, perform the following steps to join the org:
```
I'm open to other alternative phrasing here, but the current "to join an organization...to be added to the org" feels heavy and redundant.

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,82 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+For **members to join an organization**, perform the following steps to be added to the org:
+
+1. Sign in to Semgrep Cloud Platform. A **Welcome to Semgrep** dialog box appears.
+2. Check that the organization under the **Join an organization** button is the correct org to join. If there is more than 1 org, select the org to join.
```
```suggestion
2. Check that the organization under the **Join an organization** button is the correct org to join. If there is more than one org, select the org to join.

```

armchairlinguist replied:
I don't think this sentence meets any of the exception criteria in the Google style guide for small numbers, but let me know if I'm mistaken!

s-santillan replied:
I thought it fell under "**Technical quantities**, such as amounts of memory, amounts of disk space, numbers of queries, or usage limits" which is e

armchairlinguist replied:
Ah I see. I don't feel strongly about it, so feel free to ignore.

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,82 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+
+### Starting a local repository scan and sending findings to SCP
+
+You can send scan results from a local repository to Semgrep Cloud Platform. The local repository is a separate **Project** from its remote counterpart so the findings records do not overwrite each other and are kept separate.
```
I think this is usually true, but is it always true?

s-santillan replied:
There is one case I haven't verified, which is a case when someone pulls an org repo (say semgrep) then uploads findings from semgrep to the org - what happens when another user does that? I'll ask the cloud folks.

armchairlinguist replied:
I think it might overwrite in that case, potentially. My observation is it's usually `org_name/repo_name` from CI and just `repo_name` locally. 

I was also thinking about if someone goes and reads about environment variables and sets the name, but that's pretty unlikely when they're just getting started.

s-santillan replied:
Fixed in latest, I'll add a screenshot in the main comment body.

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,82 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+
+To send findings from a local repository, perform the following steps in your command-line interface:
+
+1. Login to Semgrep:
```
```suggestion
1. Log in to Semgrep:

```

armchairlinguist replied:
https://developers.google.com/style/word-list#login

I'll drop a comment on our internal example that uses `login` too :)

s-santillan replied:
Nice catch. This is one of those things I forget about when editing my work! Thank you.

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
````diff
@@ -94,6 +76,82 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+```
+
+:::info
+Many improvements to the Semgrep Cloud Platform experience only work with up-to-date Semgrep CLI versions. For this reason, Semgrep Cloud Platform only supports the 10 most recent minor versions of Semgrep CLI. For example, if the latest release was 0.114.0, all versions greater than 0.104.0 are supported while earlier versions, such as 0.103.0 can be deprecated or can result in failures.
````
```suggestion
Many improvements to the Semgrep Cloud Platform experience only work with up-to-date Semgrep CLI versions. For this reason, Semgrep Cloud Platform only supports the 10 most recent minor versions of Semgrep CLI. For example, if the latest release was 0.114.0, all versions greater than 0.104.0 are supported, while earlier versions, such as 0.103.0, can be deprecated or can result in failures.

```

armchairlinguist commented on src/components/procedure/_platform-detect-ghrepos.md:
```diff
@@ -0,0 +1,8 @@
+To ensure that your GitHub repository is **detected** by Semgrep Cloud Platform:
+
+1. Log into GitHub.
+2. Click your **profile photo > Settings > Applications**.
```
I don't think this set of steps will work for organizations as you need to select your org first. What would be the best way for us to handle that in this snippet?

s-santillan replied:
Nice catch, I created separate steps for personal vs org users :)

armchairlinguist commented:
@sachishah I wouldn't recommend that we try to lump this in with changes for cloud scanning. It will delay us from making this update, which is an important one, and make it harder to review the changes.

s-santillan commented:
@sachishah I have added some links to SSO configuration to ensure they don't fall off the user path.

<img width="674" alt="2023-08-24_02-34-55" src="https://github.com/returntocorp/semgrep-docs/assets/20766840/adc4dff2-ee91-44c5-adab-4100f98d0eba">

@armchairlinguist I added a section explaining how project slugs behave between local and remote scans

<img width="657" alt="Snag_c15dd4" src="https://github.com/returntocorp/semgrep-docs/assets/20766840/611c75c0-64f3-4062-aa8b-178697860cbc">

Thank you both for your notes, I feel that we have polished this doc quite a bit! Please approve if all good.

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,95 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+## Joining an organization
+
+:::info Prerequisites
+- Your admin or person inviting you to Semgrep Cloud Platform should have an existing organization (org) in SCP that is connected to an org in either GitHub or GitLab.
```
We may want to avoid use of the word "invite" here because folks are not actually invited. Maybe just "Your admin" should be enough?

s-santillan replied:
This is so tricky. I used the term "invite" because that's the language people used to roughly and casually describe their goal, and is the gap we're trying to address because we don't have the functionality in the product per se.

Alternatively, I could remove the above usage of "invite", then explicitly state that Semgrep does not give out email invites, and then instruct admins to tell their team members to sign in after the admin has made an org.

What do you think?

s-santillan replied:
Update: Yeah, I went in this direction :)

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,95 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+
+1. Sign in to Semgrep Cloud Platform. If your administrator has configured SSO, click **Use SSO**. A **Welcome to Semgrep** dialog box appears.
+2. Check that the organization under the **Join an organization** button is the correct org to join. If there is more than 1 org, select the org to join.
+![DESCRIPTION](/img/join-an-org.png#md-width)
```
Should DESCRIPTION still be here?

s-santillan replied:
Nice catch. Fixed in latest.

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
````diff
@@ -94,6 +76,95 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+```
+semgrep login
+```
+2. Click the login URL provided. Your are taken to your web browser to complete the login process.
````
A lot of CLIs don't allow a direct click on this link - but not sure if that distinction really matters?

s-santillan replied:
Click or copy. In the latest client, I was amazed that my WSL actually automatically opened my windows program. I am truly living in the future, 2 OS in 1. Cygwin could never.

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
````diff
@@ -94,6 +76,95 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+semgrep ci
+```
+
+#### Projects separation between local and remote repositories
````
```suggestion
#### Project separation between local and remote repositories
```
Projects sounds a bit odd to me. If we do change this, the link above would need to be changed too.

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,95 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+
+#### Projects separation between local and remote repositories
+
+The Project slug for a remote repository always takes the form `account-name/repository-name`. The Project slug for a local repository always takes the form `repository-name`. Refer to the following screenshot for an example of both remote and local Projects in a single personal account.
```
```suggestion
The Project slug for a remote repository takes the form `account-name/repository-name`. The Project slug for a local repository takes the form `repository-name`. Refer to the following screenshot for an example of both remote and local Projects in a single personal account.
```
we could add "by default", but mostly I just don't think it makes sense to say "always" even if it's just in a getting started page, since it's a configurable value.

s-santillan replied:
Makes sense. I think removing the word "always" or "never" solves it :) Thank you.

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,95 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+
+The Project slug for a remote repository always takes the form `account-name/repository-name`. The Project slug for a local repository always takes the form `repository-name`. Refer to the following screenshot for an example of both remote and local Projects in a single personal account.
+
+* **For personal accounts:** A local repository scan **never** overwrites findings records of its remote counterpart. They are two separate Projects. Personal accounts only have one team member or user: you.
```
```suggestion
* **For personal accounts:** A local repository scan does not overwrite the findings records of its remote counterpart. They are two separate Projects. Personal accounts only have one team member or user: you.
```

armchairlinguist commented on docs/semgrep-cloud-platform/getting-started.md:
```diff
@@ -94,6 +76,95 @@ Using Semgrep Supply Chain with Semgrep Cloud Platform provides users with the V
+The Project slug for a remote repository always takes the form `account-name/repository-name`. The Project slug for a local repository always takes the form `repository-name`. Refer to the following screenshot for an example of both remote and local Projects in a single personal account.
+
+* **For personal accounts:** A local repository scan **never** overwrites findings records of its remote counterpart. They are two separate Projects. Personal accounts only have one team member or user: you.
+* For **organization accounts**: A local repository scan never overwrites findings records of its remote counterpart, but if two members both send local repository findings, their records can overwrite each other's findings. This is because org accounts can have more than one team member, but all local scans are sent to the same Project slug.
```
```suggestion
* For **organization accounts**: A local repository scan does not overwrite the findings records of its remote counterpart, but if two members both send local repository findings, their records can overwrite each other's findings. This is because org accounts can have more than one team member, but all local scans are sent to the same Project slug.
```
"their records can overwrite each other's findings" sounds weird here but I can't think of a great replacement. If you can, though, I'm for using it :)

s-santillan replied:
You're right, it's awkward. 

Alternative: if two members both send local repository findings, one set of results may overwrite the other unintentionally.

armchairlinguist reviewed (approved):
This is a huge improvement for sure, thank you for all the hard work, @s-santillan ! I still had a few comments, mainly about the use of always/never about project names, but I leave it to your discretion.
